src/mapogcsld.cpp

```diff
@@ -196,6 +249,206 @@ static void msSLDApplySLD_DuplicateLayers(mapObj *map, int nSLDLayers,
 }
 #endif
 
+int msApplySldLayerToMapLayer(layerObj *sldLayer, layerObj *lp) {
+
+  if (sldLayer->numclasses > 0) {
+    int iClass = 0;
+    int k;
+    int bSLDHasNamedClass = MS_FALSE;
+
+    lp->type = sldLayer->type;
+    lp->rendermode = MS_ALL_MATCHING_CLASSES;
+
+    for (k = 0; k < sldLayer->numclasses; k++) {
+      if (sldLayer->_class[k]->group) {
+        bSLDHasNamedClass = MS_TRUE;
+        break;
+      }
+    }
+
+    for (k = 0; k < lp->numclasses; k++) {
+      if (lp->_class[k] != NULL) {
+        lp->_class[k]->layer = NULL;
+        if (freeClass(lp->_class[k]) == MS_SUCCESS) {
+          msFree(lp->_class[k]);
+          lp->_class[k] = NULL;
+        }
+      }
+    }
+    lp->numclasses = 0;
+
+    if (bSLDHasNamedClass && sldLayer->classgroup) {
+      /* Set the class group to the class that has UserStyle.IsDefault
+       */
+      msFree(lp->classgroup);
+      lp->classgroup = msStrdup(sldLayer->classgroup);
+    } else {
+      /*unset the classgroup on the layer if it was set. This allows the
+      layer to render with all the classes defined in the SLD*/
+      msFree(lp->classgroup);
+      lp->classgroup = NULL;
+    }
+
+    for (k = 0; k < sldLayer->numclasses; k++) {
+      if (msGrowLayerClasses(lp) == NULL)
+        return MS_FAILURE;
+
+      initClass(lp->_class[iClass]);
+      msCopyClass(lp->_class[iClass], sldLayer->_class[k], NULL);
+      lp->_class[iClass]->layer = lp;
+      lp->numclasses++;
+
+      /*aliases may have been used as part of the sld text symbolizer
+        for label element. Try to process it if that is the case #3114*/
+
+      int layerWasOpened = msLayerIsOpen(lp);
+
+      if (layerWasOpened || msLayerOpen(lp) == MS_SUCCESS) {
+        if (layerWasOpened || msLayerGetItems(lp) == MS_SUCCESS) {
+
+          if (lp->_class[iClass]->text.string) {
+            int z;
+            for (z = 0; z < lp->numitems; z++) {
+              const char *pszFullName;
+              char szTmp[512];
+              char *pszTmp1;
+              if (!lp->items[z] || strlen(lp->items[z]) == 0)
+                continue;
+              snprintf(szTmp, sizeof(szTmp), "%s_alias", lp->items[z]);
+              pszFullName = msOWSLookupMetadata(&(lp->metadata), "G", szTmp);
+              pszTmp1 = msStrdup(lp->_class[iClass]->text.string);
+              if (pszFullName != NULL &&
+                  (strstr(pszTmp1, pszFullName) != NULL)) {
+                pszTmp1 =
+                    msReplaceSubstring(pszTmp1, pszFullName, lp->items[z]);
+                char *pszTmp2 =
+                    (char *)malloc(sizeof(char) * (strlen(pszTmp1) + 3));
+                sprintf(pszTmp2, "(%s)", pszTmp1);
+                msLoadExpressionString(&(lp->_class[iClass]->text), pszTmp2);
+                msFree(pszTmp2);
```

Comment:
```suggestion
                msLoadExpressionString(&(lp->_class[iClass]->text), std::string("(").append(pszTmp1).append(")").c_str());
```

Reply:
Updated, but `msLoadExpressionString` expects `char *` and not `const char *`. Changing this would impact lots of places in the code so probably outside the scope of this pull request. I used the following construct to cast the c_str, but not sure if this is correct:

```
              char *pszTmp2 = const_cast<char *>(
                  std::string("(").append(pszTmp1).append(")").c_str());
```

Reply:
> Updated, but `msLoadExpressionString` expects `char *` and not `const char *`

should be fixed per https://github.com/MapServer/MapServer/pull/7035 . You can rebase on top of latest master once this will be merged